spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkWriteConf.java

```diff
@@ -262,7 +278,50 @@ private DistributionMode defaultWriteDistributionMode() {
     }
   }
 
-  public DistributionMode deleteDistributionMode() {
+  public SparkWriteRequirements copyOnWriteRequirements(Command command) {
+    if (ignoreTableDistributionAndOrdering()) {
+      LOG.info("Skipping distribution/ordering: disabled per job configuration");
+      return SparkWriteRequirements.EMPTY;
+    }
+
+    return SparkWriteUtil.copyOnWriteRequirements(
+        table, command, copyOnWriteDistributionMode(command), fanoutWriterEnabled());
+  }
+
+  @VisibleForTesting
+  DistributionMode copyOnWriteDistributionMode(Command command) {
+    switch (command) {
+      case DELETE:
+        return deleteDistributionMode();
+      case UPDATE:
+        return updateDistributionMode();
+      case MERGE:
+        return copyOnWriteMergeDistributionMode();
+      default:
+        throw new IllegalArgumentException("Unexpected command: " + command);
+    }
+  }
+
+  public SparkWriteRequirements positionDeltaRequirements(Command command) {
+    return SparkWriteUtil.positionDeltaRequirements(
```

Comment:
Can't skip distribution and ordering because the spec requires position deletes to be sorted.

Reply:
I have a follow-up PR that adds a fanout position delete writer. It buffers deletes into a bitmap and then produces a sorted file when closed. In the future, we either remove the requirement for position deletes to be sorted or add support for Puffin delete files that would persist bitmaps.

Reply:
That's beyond this release, though.